### DISCUSSION TITLE
Atoms related to the matrix logarithm

### DISCUSSION THREAD

**Discussion by akshayka** [cvxpy/cvxpy#1275] [Ideas]
A community member suggested that CVXPY could benefit from implementing atoms related to the matrix logarithm, such as the ones implemented in this CVX package (https://github.com/hfawzi/cvxquad) and described in this paper (https://arxiv.org/pdf/1705.00812.pdf).

If you're interested in these atoms, let us know by reacting with a :+1:  emoji and/or dropping a note in the comments below.

**Comment by Mark-L-Stone** [comment]
![image](https://user-images.githubusercontent.com/23507598/110972146-acbe9100-8329-11eb-9498-b6f5f2c3447b.png)
Very much a nontrivial endeavor, but valuable. Beware though, that as powerful as Quantum Relative Entropy is, implementing by the approach in the paper https://arxiv.org/pdf/1705.00812.pdf) leads to large SDPs, particularly for Quantum Relative Entropy, less so for other functions and sets (see the table at the top of p.3). Unfortunately, there is no currently known self-concordant barrier function for operator relative entropy cone which allow for a more favorable computation than the approach in the paper.

Here are some problems which could be handled with the proposed functionality. http://ask.cvxr.com/t/perspective-of-log-det-function-and-cvx-formulations-of-related-log-det-problems-using-quantum-relative-entropy-from-cvxquad/4033

**Comment by rileyjmurray** [comment]
I'm definitely interested in this. Unfortunately, I won't have the bandwidth to take the lead here for the foreseeable future. One thing of note: the hypograph of the von Neumann entropy (``-trace(X @ log(X))``) can be represented with the exponential cone and PSD cone. So we can handle that one with existing CVXPY-supported constraints. Von Neumann entropy on its own has some nice theoretical applications in kernel methods.
